pkg/k8sutil/k8sutil.go

```diff
@@ -383,3 +385,45 @@ func mergeMapsByPrefix(from map[string]string, to map[string]string, prefix stri
 
 	return to
 }
+
+// DiffRulerConfigMap preProvisionRuleConfigMapNumber can't less than newConfigMaps number.
+func DiffRulerConfigMap(currentConfigMaps []v1.ConfigMap, newConfigMaps []v1.ConfigMap) (createOrUpdateConfigMaps []v1.ConfigMap, deleteConfigMaps []v1.ConfigMap) {
+	sort.Slice(newConfigMaps, func(i, j int) bool {
+		return newConfigMaps[i].Name < newConfigMaps[j].Name
+	})
+
+	sort.Slice(currentConfigMaps, func(i, j int) bool {
+		return currentConfigMaps[i].Name < currentConfigMaps[j].Name
+	})
+	currentConfigMapsNum := len(currentConfigMaps)
+	newConfigMapsNum := len(newConfigMaps)
+	if newConfigMapsNum >= currentConfigMapsNum {
+		createOrUpdateConfigMaps = newConfigMaps[0:newConfigMapsNum]
+	}
+
+	if newConfigMapsNum < currentConfigMapsNum {
+		deleteConfigMaps = currentConfigMaps[newConfigMapsNum:currentConfigMapsNum]
+		createOrUpdateConfigMaps = newConfigMaps
+	}
+
+	return createOrUpdateConfigMaps, deleteConfigMaps
+}
+
+func CreateOrUpdateConfigMap(ctx context.Context, client clientv1.ConfigMapInterface, configMap *v1.ConfigMap) error {
```

Comment:
I'm not sure about this, but here couldn't we use [1] instead of maintaining this code. @simonpasquier might know what's best to do in this specific case.

[1] https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrUpdate

Reply:
I don't think it's good to use this method? If yes, why not all create resources using this package.